### PR TITLE
Feat/event tracking

### PR DESCRIPTION
**Pull request by myo3**

### CODE REVIEW

File: Swift Demo App/iOS-SDK-Demo/ALEventTrackingViewController.swift [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int
+    {
+        return events.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell
+    {
+        var cell = tableView.dequeueReusableCell(withIdentifier: "rootPrototype")
+        if(cell == nil){
```
Comment (christophercong) [inline Swift Demo App/iOS-SDK-Demo/ALEventTrackingViewController.swift]: Spacing.

File: Swift Demo App/iOS-SDK-Demo/ALEventTrackingViewController.swift [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    // MARK: Table View Data Source
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int
+    {
+        return events.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell
+    {
+        var cell = tableView.dequeueReusableCell(withIdentifier: "rootPrototype")
```
Comment (christophercong) [inline Swift Demo App/iOS-SDK-Demo/ALEventTrackingViewController.swift]: I recommend using `tableView.dequeueReusableCell(withIdentifier: "rootPrototype", for: indexPath)`. It'll help you avoid returning an explicitly unwrapped optional.

File: Swift Demo App/iOS-SDK-Demo/ALEventTrackingViewController.swift [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        return cell!
+    }
+    
+    // MARK: Table View Delegate
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath)
+    {
+        super.tableView(tableView, didSelectRowAt: indexPath)
+        
+        let eventService: ALEventService = (ALSdk.shared()?.eventService)!
```
Comment (christophercong) [inline Swift Demo App/iOS-SDK-Demo/ALEventTrackingViewController.swift]: Forgive me, but I am a huge proponent in handling all optionals (and never using `!`). We should use a guard here and exit early (and update title) if eventService is nil.

File: Objective-C Demo App/iOS-SDK-Demo/ALEventTrackingViewController.m [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        case 4: {
+            [eventService trackEvent: kALEventTypeUserCompletedLevel
+                          parameters: @{kALEventParameterCompletedLevelKey : @"LEVEL NAME OR NUMBER"}];
+            break;
+        }
+        case 5: {
+            [eventService trackEvent: kALEventTypeUserCreatedReservation
+                          parameters: @{kALEventParameterProductIdentifierKey    : @"PRODUCT SKU OR ID",
+                                        kALEventParameterReservationStartDateKey : @"START NSDATE",
+                                        kALEventParameterReservationEndDateKey   : @"END NSDATE"}];
```
Comment (thomasmso) [inline Objective-C Demo App/iOS-SDK-Demo/ALEventTrackingViewController.m]: I think this is an error on the documentation, `NSDate` doesn't make sense.

**Pull request merged**